### ISSUE TITLE
sysrepocfg --rpc=get-config.xml not returning data for plugin callback

### ISSUE THREAD

**Issue by csc-gip** [sysrepo/sysrepo#3046]
Hi! I think I am missing something.
I implemented a test-plugin and successfully subscribed to
```
        rc = sr_rpc_subscribe_tree(my_session, "/ietf-netconf:get-config", get_config_cb, NULL, 0, SR_SUBSCR_DEFAULT, &my_subscription);
```
I am getting data (from running)
```
        sr_get_data(session, "/*", 0, 0, SR_GET_NO_FILTER | SR_OPER_DEFAULT, &data);

```
and setting output of the rpc callback
```
        if (data != NULL)
        {
            struct lyd_node * node = NULL;
            rc = lyd_new_any(output, NULL, "data", data->tree, 1, LYD_ANYDATA_DATATREE, 1, &node);
            SRPLG_LOG_DBG(PLG_NAME, "node inserted: %u %s\n", rc, ly_errmsg(sr_acquire_context(sr_session_get_connection(session))));
            data->tree = NULL;
            sr_release_data(data);
        }
```
I am not getting an output from sysrepocfg
```
sysrepocfg -v3 --rpc=get-config.xml
[INF] Connection 19 created.
[INF] Session 36 (user "vscode", CID 19) created.
[INF] EV ORIGIN: "/ietf-netconf:get-config" "rpc" ID 1 priority 0 for 1 subscribers published.
[INF] EV ORIGIN: "/ietf-netconf:get-config" "rpc" ID 1 priority 0 succeeded.
```
When printing output in the callback function, before returning
```
        lyd_print_mem(&strp, output, LYD_XML, LYD_PRINT_WD_ALL | LYD_PRINT_WITHSIBLINGS);
```
I get
```
[DBG] test-plugin: node inserted: 0 (null)

[DBG] test-plugin: get-config done, returning: 
[DBG] test-plugin: <get-config xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <data>
    <nacm xmlns="urn:ietf:params:xml:ns:yang:ietf-netconf-acm">
      <enable-nacm>true</enable-nacm>
      <read-default>permit</read-default>
      <write-default>deny</write-default>
      <exec-default>permit</exec-default>
      <enable-external-groups>true</enable-external-groups>
    </nacm>
  </data>
</get-config>

[INF] EV LISTEN: "/ietf-netconf:get-config" "rpc" ID 1 priority 0 success (remaining 0 subscribers).
```
I am using sysrepo devel without any additional modules installed or changes to datastores.

**Comment by michalvasko** [comment]
Based on everything you posted here I believe the data you are setting in the output are all default nodes. You see them in your debug print because of using the flag `LYD_PRINT_WD_ALL` but *sysrepocfg* is not using this flag so it does not print anything.

**Comment by csc-gip** [comment]
Thanks a lot. I added some data and removed the `LYD_PRINT_WD_ALL` flag. Still no difference.
```
[DBG] test-plugin: <get-config xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <data>
    <hardware xmlns="urn:ietf:params:xml:ns:yang:ietf-hardware">
      <component>
        <name>main 0</name>
        <class xmlns:ianahw="urn:ietf:params:xml:ns:yang:iana-hardware">ianahw:module</class>
      </component>
    </hardware>
  </data>
</get-config>
````
What confueses me is that output already contains `<get-config>` as toplevel node, when the callback gets called. i.e. output is not empty on calback entry and it contains the input tag.

Some things I should have mentioned.
The plugin is running as standalone programm (direct approach) for testing.
I also opened a new session in `sr_plugin_init_cb` as metioned here: https://www.sysrepo.org/plugins/documentation
Not opening a new session does not help, unfortunately.
